solr/core/src/test/org/apache/solr/cloud/ReplicateFromLeaderTest.java

```diff
@@ -28,44 +28,64 @@ public class ReplicateFromLeaderTest {
     assertEquals("0:0:7", pollInterval);
 
     updateHandlerInfo =
         new SolrConfig.UpdateHandlerInfo(
-            "solr.DirectUpdateHandler2", -1, 60000, -1, true, -1, 15000, false);
+            "solr.DirectUpdateHandler2", -1, 60000, -1, true, -1, 15000, false, null);
     pollInterval = ReplicateFromLeader.determinePollInterval(updateHandlerInfo);
     assertEquals("0:0:30", pollInterval);
 
     updateHandlerInfo =
         new SolrConfig.UpdateHandlerInfo(
-            "solr.DirectUpdateHandler2", -1, 15000, -1, false, -1, 60000, false);
+            "solr.DirectUpdateHandler2", -1, 15000, -1, false, -1, 60000, false, null);
     pollInterval = ReplicateFromLeader.determinePollInterval(updateHandlerInfo);
     assertEquals("0:0:30", pollInterval);
 
     updateHandlerInfo =
         new SolrConfig.UpdateHandlerInfo(
-            "solr.DirectUpdateHandler2", -1, 60000, -1, false, -1, 15000, false);
+            "solr.DirectUpdateHandler2", -1, 60000, -1, false, -1, 15000, false, null);
     pollInterval = ReplicateFromLeader.determinePollInterval(updateHandlerInfo);
     assertEquals("0:0:30", pollInterval);
 
     updateHandlerInfo =
         new SolrConfig.UpdateHandlerInfo(
-            "solr.DirectUpdateHandler2", -1, -1, -1, false, -1, 60000, false);
+            "solr.DirectUpdateHandler2", -1, -1, -1, false, -1, 60000, false, null);
     pollInterval = ReplicateFromLeader.determinePollInterval(updateHandlerInfo);
     assertEquals("0:0:30", pollInterval);
 
     updateHandlerInfo =
         new SolrConfig.UpdateHandlerInfo(
-            "solr.DirectUpdateHandler2", -1, 15000, -1, false, -1, -1, false);
+            "solr.DirectUpdateHandler2", -1, 15000, -1, false, -1, -1, false, null);
     pollInterval = ReplicateFromLeader.determinePollInterval(updateHandlerInfo);
     assertEquals("0:0:7", pollInterval);
 
     updateHandlerInfo =
         new SolrConfig.UpdateHandlerInfo(
-            "solr.DirectUpdateHandler2", -1, 60000, -1, true, -1, -1, false);
+            "solr.DirectUpdateHandler2", -1, 60000, -1, true, -1, -1, false, null);
     pollInterval = ReplicateFromLeader.determinePollInterval(updateHandlerInfo);
     assertEquals("0:0:30", pollInterval);
+
+    updateHandlerInfo =
+        new SolrConfig.UpdateHandlerInfo(
+            "solr.DirectUpdateHandler2", -1, 60000, -1, true, -1, -1, false, "0:0:56");
+    pollInterval = ReplicateFromLeader.determinePollInterval(updateHandlerInfo);
+    assertEquals("0:0:56", pollInterval);
+
+    updateHandlerInfo =
+        new SolrConfig.UpdateHandlerInfo(
+            "solr.DirectUpdateHandler2",
+            -1,
+            60000,
+            -1,
+            true,
+            -1,
+            -1,
+            false,
+            "garbage-unfortunately");
+    pollInterval = ReplicateFromLeader.determinePollInterval(updateHandlerInfo);
+    assertEquals("garbage-unfortunately", pollInterval);
```

Comment:
Perhaps `ReplicationHander.readIntervalNs` could be made public and used to validate either in UpdateHandlerInfo or in determinePollInterval. With `expectThrows(...)` in the test logic here then.

Reply:
Fail early, fail fast it is